spec/patches.md

```diff
@@ -28,28 +28,58 @@ a specific [=task queue=].
 With: For each [=event loop=], every [=task source=] that is not a
 [=scheduler task source=] must be associated with a specific [=task queue=].
 
+An [=event loop=] object has a numeric <dfn for="event loop">next enqueue order</dfn>, which is
+is initialized to 1.
+
+Note: The [=event loop/next enqueue order=] is a strictly increasing number that is used to
+determine task execution order across [=scheduler task queues=] of the same {{TaskPriority}} across
+all {{Scheduler}}s associated with the same [=event loop=]. A timestamp would also suffice as long
+as it is guaranteed to be strictly increasing and unique.
+
 ### <a href="https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-processing-model">Event loop: processing model</a> ### {#sec-patches-html-event-loop-processing}
 
-Add the following steps to the event loop processing steps, before step 1:
-
-  1. Let |queues| be the [=set=] of the [=event loop=]'s [=task queues=] that
-     contain at least one <a for="task">runnable</a> <a for="/">task</a>.
-  1. Let |schedulers| be the [=set=] of all {{Scheduler}} objects whose
-     [=relevant agent's=] [=agent/event loop=] is this event loop and that
-     [=have a runnable task=].
-  1. If |schedulers| and |queues| are both [=list/empty=], skip to the
-     <code>microtasks</code> step below.
-
-Modify step 1 to read:
-
-  1. Let |taskQueue| be one of the following, chosen in an
-     [=implementation-defined=] manner:
-    * If |queues| is not [=list/empty=], one of the [=task queues=] in |queues|,
-       chosen in an [=implementation-defined=] manner.
-    * If |schedulers| is not [=list/empty=], the result of
-       [=selecting the task queue of the next scheduler task=] from one of the
-       {{Scheduler}}s in |schedulers|, chosen in an [=implementation-defined=]
-       manner.
+Replace:
+ 1. If the [=event loop=] has a [=task queue=] with at least one <a for="task">runnable</a>
+    <a for="/">task</a>, then:
+   1. Let |taskQueue| be one such [=task queue=], chosen in an [=implementation-defined=] manner.
+
+With:
+  1. Let |queues| be the [=set=] of the [=event loop=]'s [=task queues=] that contain at least one
+     <a for="task">runnable</a> <a for="/">task</a>.
+  1. Let |scheduler queue| be the result of [=selecting the next scheduler queue from all schedulers=]
+     given the [=event loop=].
+  1. If |scheduler queue| is not null and |queues| is not [=list/empty=], then:
+    1. If |scheduler queue|'s [=scheduler task queue/priority=] is {{TaskPriority/user-blocking}},
+       then [=set/remove=] from |queues| any [=task queue=] whose [=task source=] is in
+       «[=timer task source=], [=posted message task source=]».
```

Comment:
This will delay behind *all* posted messages, not just same-Window ones. The below note doesn't seem to align with that.

If you wanted to restrict to same-window ones, you'd need to mark the queued tasks in some way, presumably at the time they're enqueued. I might try something like moneypatching https://html.spec.whatwg.org/multipage/web-messaging.html#posted-message-task-source to contain "If ..., the [task] queued in such a way is a **same-window posted message task**", and then checking that here.